src/porepy/numerics/ad/operators.py

```diff
@@ -1431,6 +1465,30 @@ def shape(self) -> tuple[int, int]:
         """Shape of the wrapped matrix."""
         return self._shape
 
+    @staticmethod
+    def _compute_spmatrix_hash(mat: sps.spmatrix) -> str:
+        """Utility function that handles all the sparse formats to compute the hash."""
+        # proper handling of all the formats like csr, coo, etc..
+        if isinstance(
+            mat,
+            (
+                sps.csr_matrix,
```

Comment:
This will fail for dia arrays / matrices. Is there a way to do this without having to explicitly treat the various types matrices?